DependencyInjection/LexikJWTAuthenticationExtension.php

```diff
@@ -186,4 +189,37 @@ private static function createTokenExtractors(ContainerBuilder $container, array
+            $container->setParameter('lexik_jwt_authentication.web_token.builder.signature_key', $config['access_token_issuance']['signature']['key']);
+            if ($config['access_token_issuance']['encryption']['enabled'] === true) {
+                $container->setParameter('lexik_jwt_authentication.web_token.builder.key_encryption_algorithm', $config['access_token_issuance']['encryption']['key_encryption_algorithm']);
+                $container->setParameter('lexik_jwt_authentication.web_token.builder.content_encryption_algorithm', $config['access_token_issuance']['encryption']['content_encryption_algorithm']);
+                $container->setParameter('lexik_jwt_authentication.web_token.builder.encryption_key', $config['access_token_issuance']['encryption']['key']);
+            }
+            $loader->load('web_token_issuance.xml');
+        }
+        if ($config['access_token_verification']['enabled'] === true) {
+            $container->setParameter('lexik_jwt_authentication.web_token.loader.allowed_signature_algorithms', $config['access_token_verification']['signature']['allowed_signature_algorithms']);
+            $container->setParameter('lexik_jwt_authentication.web_token.loader.signature_keyset', $config['access_token_verification']['signature']['keyset']);
+            $container->setParameter('lexik_jwt_authentication.web_token.loader.claim_checkers', $config['access_token_verification']['signature']['claim_checkers']);
+            $container->setParameter('lexik_jwt_authentication.web_token.loader.jws_header_checkers', $config['access_token_verification']['signature']['header_checkers']);
+            $container->setParameter('lexik_jwt_authentication.web_token.loader.mandatory_claims', $config['access_token_verification']['signature']['mandatory_claims']);
+            if ($config['access_token_verification']['encryption']['enabled'] === true) {
+                $container->setParameter('lexik_jwt_authentication.web_token.loader.continue_on_decryption_failure', $config['access_token_verification']['encryption']['continue_on_decryption_failure']);
+                $container->setParameter('lexik_jwt_authentication.web_token.loader.jwe_header_checkers', $config['access_token_verification']['encryption']['header_checkers']);
+                $container->setParameter('lexik_jwt_authentication.web_token.loader.allowed_key_encryption_algorithms', $config['access_token_verification']['encryption']['allowed_key_encryption_algorithms']);
+                $container->setParameter('lexik_jwt_authentication.web_token.loader.allowed_content_encryption_algorithms', $config['access_token_verification']['encryption']['allowed_content_encryption_algorithms']);
+                $container->setParameter('lexik_jwt_authentication.web_token.loader.encryption_keyset', $config['access_token_verification']['encryption']['keyset']);
```

Comment:
We should avoid creating parameters and call  `replaceArgument()` on the relevant definitions instead to avoid cluttering the cached container with user-configured values.

Reply:
Make sense. I replaced all parameters in favor of `replaceArgument()` calls